Firestore/core/src/firebase/firestore/nanopb/writer.h

```diff
@@ -66,17 +63,43 @@ class Writer {
 class ByteStringWriter : public Writer {
  public:
   ByteStringWriter();
+  ~ByteStringWriter();
+
+  ByteStringWriter(const ByteStringWriter&) = delete;
+  ByteStringWriter(ByteStringWriter&&) = delete;
+
+  ByteStringWriter& operator=(const ByteStringWriter&) = delete;
+  ByteStringWriter& operator=(ByteStringWriter&&) noexcept = delete;
 
-  nanopb::ByteString ToByteString() const;
+  void Append(const uint8_t* data, size_t size);
+  void Reserve(size_t size);
```

Comment:
`reserve` on `std::vector` takes the _total_ desired capacity, while this function takes a delta applied to the current size. Please call it out in a comment (or perhaps just change the behavior to mimic `vector`).

Reply:
If there are more users of this API we can revisit. For now, matching `std::vector::reserve` is fine. I could change the name to `AddCapacity` if I wanted to change to the other model.